packages/bundle-analyzer/src/cli.ts

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env node
+
+import path from "node:path";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import { createAndUploadReport } from "./index.js";
+import { red, type Options } from "@codecov/bundler-plugin-core";
+
+interface Argv {
+  buildDirectoryPath: string;
+  dryRun: boolean;
+  uploadToken?: string;
+  apiUrl: string;
+  bundleName?: string;
+  debug: boolean;
+}
+
+const argv = yargs(hideBin(process.argv))
+  .usage("Usage: $0 <build-directory-path> [options]")
+  .command(
+    "$0 <build-directory-path>",
+    "Analyze and upload bundle report",
+    (yargs) => {
+      return yargs.positional("buildDirectoryPath", {
+        describe: "The path to the build directory",
+        type: "string",
+        demandOption: true,
+      });
+    },
+  )
+  .options({
+    "dry-run": {
+      alias: "d",
+      type: "boolean",
+      description: "Perform a dry run without uploading",
+      default: false,
+    },
+    "upload-token": {
+      alias: "t",
+      type: "string",
+      description: "Specify the upload token for authentication",
+    },
+    "api-url": {
+      alias: "a",
+      type: "string",
+      description: "Set the API URL",
+      default: "https://api.codecov.io",
+    },
+    "bundle-name": {
+      alias: "n",
+      type: "string",
+      description: "Set the bundle identifier in Codecov",
+      demandOption: true,
+    },
+    debug: {
+      alias: "v",
+      type: "boolean",
+      description: "Enable debug mode for additional logging",
+      default: false,
+    },
+  })
+  .strict()
+  .help("h")
+  .alias("h", "help")
+  .parseSync() as unknown as Argv;
+
+function prepareCoreOptions(): Options {
+  const uploadToken = argv.uploadToken ?? process.env.CODECOV_UPLOAD_TOKEN;
+  if (!uploadToken) {
+    throw new Error(
+      "An upload token is required. Use --upload-token or set the CODECOV_UPLOAD_TOKEN environment variable.",
+    );
+  }
```

Comment:
Do you want to add in a TODO here, or note it down somewhere that we'll need to update these lines here to add in support for tokenless and GH OIDC.

Reply:
Good call-out! I'll track it in this TODO list to updated it here (and everywhere - READMEs, etc.)
https://github.com/codecov/engineering-team/issues/2464